From 9074be0bc7d336f50648b88d0478ce5efa37f3c1 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Tue, 24 May 2022 07:56:55 -0700
Subject: [PATCH 1/3] ath6kl: Use cc-disable-warning to disable
 -Wdangling-pointer

Clang does not support this option so the build fails:

  error: unknown warning option '-Wno-dangling-pointer' [-Werror,-Wunknown-warning-option]

Use cc-disable-warning so that the option is only added when it is
supported.

Fixes: bd1d129daa3e ("wifi: ath6k: silence false positive -Wno-dangling-pointer warning on GCC 12")
Reported-by: "kernelci.org bot" <bot@kernelci.org>
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/20220524145655.869822-1-nathan@kernel.org
---
 drivers/net/wireless/ath/ath6kl/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath6kl/Makefile b/drivers/net/wireless/ath/ath6kl/Makefile
index 01cc0d50fee6..a75bfa9fd1cf 100644
--- a/drivers/net/wireless/ath/ath6kl/Makefile
+++ b/drivers/net/wireless/ath/ath6kl/Makefile
@@ -38,7 +38,7 @@ ath6kl_core-y += recovery.o
 
 # FIXME: temporarily silence -Wdangling-pointer on non W=1+ builds
 ifndef KBUILD_EXTRA_WARN
-CFLAGS_htc_mbox.o += -Wno-dangling-pointer
+CFLAGS_htc_mbox.o += $(call cc-disable-warning, dangling-pointer)
 endif
 
 ath6kl_core-$(CONFIG_NL80211_TESTMODE) += testmode.o
-- 
2.36.1

